src/utils/api.ts

```diff
@@ -33,6 +34,22 @@ export class ApiService {
     this.app.get("/", (_req: Request, res: Response) => {
       res.send("🐮 Moooo!");
     });
+    this.app.get("/health", async (_req: Request, res: Response) => {
+      // Using an iterator, process all the chain contexts, storing the health
+      // in a map, and if any of the contexts are unhealthy, return false
+      const contexts = Object.values(ChainContext.chains) as ChainContext[];
+      const healths = contexts.map((context) => context.health);
+      const healthStatusDict = healths.reduce((acc, health) => {
+        acc[health.chainId] = health;
+        return acc;
+      }, {} as { [chainId: string]: ChainHealth });
+
+      const overallHealth = healths.every((health) => health.isHealthy);
+
+      res
+        .status(overallHealth ? 200 : 500)
```

Comment:
I'm not sure 500 is the best for this situation.
After all, the service is able to handle the request and prepare a response without erroing internally.
It's just that some parts of the system aren't healthy.

Eliciting help from SO, there are the 2 interpretations in the top 2 answers: https://stackoverflow.com/questions/25389261/which-http-status-code-should-i-use-for-a-health-check-failure

Since additional info is sent as part of the payload, I'd say this should be used to determine the service status and if the service is able to respond successfully, it should be a 200 status code.

Reply:
Got it.
Let's go with 5xx then.
What about 503, as pointed out in [this answer](https://stackoverflow.com/a/48005358)?

Reply:
Sounds great!

Reply:
Done.